odl/set/sets.py

```diff
@@ -217,6 +237,15 @@ def __contains__(self, other):
         characters, `False` otherwise."""
         return isinstance(other, basestring) and len(other) == self.length
 
+    def contains_all(self, array):
+        """Test if `array` is an array of strings with correct length."""
```

Comment:
And the default implementation is, in a way, hidden in `np.result_type(*array)`, but the rest of the code is an optimization for NumPy arrays.